plugins/available/alias-completion.plugin.bash

```diff
@@ -63,14 +63,17 @@ function alias_completion {
             local compl_func="${new_completion/#* -F /}"; compl_func="${compl_func%% *}"
             # avoid recursive call loops by ignoring our own functions
             if [[ "${compl_func#_$namespace::}" == $compl_func ]]; then
+                if [[ $compl_func == _filedir_xspec ]]; then
+                    compl_func_args=$alias_cmd
+                fi
                 local compl_wrapper="_${namespace}::${alias_name}"
                     echo "function $compl_wrapper {
                         (( COMP_CWORD += ${#alias_arg_words[@]} ))
                         COMP_WORDS=($alias_cmd $alias_args \${COMP_WORDS[@]:1})
                         (( COMP_POINT -= \${#COMP_LINE} ))
                         COMP_LINE=\${COMP_LINE/$alias_name/$alias_cmd $alias_args}
                         (( COMP_POINT += \${#COMP_LINE} ))
-                        $compl_func
+                        $compl_func $compl_func_args
```

Comment:
Potentially quote the `$compl_func_args` usage here? Is this always a single value, or can this also be an array?

Reply:
I would not exactly know the argument is always single or takes in an array... I know in the case of `_filedir_xspec` a single parameter of the shell command is required, but in general it may depend on the specific completion function. I have not done exhaustive search for the function signature patterns for all the completion functions, and that's why I have coded in `compl_func_args` as optional. In a couple of cases I have seen them taking in argument that is different from what `_filedir_xspec` expects, so it is likely we need to take care of this case by case.

Quoting `$compl_func_args` makes it a single parameter, which may not be desired in general. Leaving it in without quotes, parameter array can be stored in `compl_func_args` without modification of that line.

Reply:
OK, thanks for the details! Let's leave it like that for the time being.